nilearn/image/image.py

```diff
@@ -638,6 +641,9 @@ def threshold_img(img, threshold, mask_img=None):
     img_data = _safe_get_data(img).copy()
```

Comment:
Not related to your PR but this looks suspicious to me: there is no need to copy data coming from `_safe_get_data`.